dev/com.ibm.ws.security.registry.basic_fat/fat/src/com/ibm/ws/security/registry/basic/fat/FATTest.java

```diff
@@ -18,63 +18,75 @@
 import org.junit.runner.RunWith;
 
 import com.ibm.websphere.simplicity.log.Log;
 import com.ibm.ws.security.registry.SearchResult;
 import com.ibm.ws.security.registry.test.UserRegistryServletConnection;
 
+import componenttest.annotation.CheckpointTest;
 import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.CheckpointRule;
+import componenttest.rules.repeater.CheckpointRule.ServerMode;
 import componenttest.topology.impl.LibertyServer;
 import componenttest.topology.impl.LibertyServerFactory;
 import componenttest.vulnerability.LeakedPasswordChecker;
 
 @RunWith(FATRunner.class)
+@CheckpointTest(alwaysRun = true)
 public class FATTest {
     private static final String DEFAULT_CONFIG_FILE = "basic.server.xml.orig";
     private static final String ALTERNATE_BASIC_REGISTRY_CONFIG = "alternateBasicRegistry.xml";
     private static final String DEFAULT_AES_CONFIG_FILE = "defaultAESBasicRegistry.xml";
     private static final String CUSTOM_AES_CONFIG_FILE = "customAESBasicRegistry.xml";
     private static final String DEFAULT_HASH_CONFIG_FILE = "defaultHashBasicRegistry.xml";
     private static String serverConfigurationFile = DEFAULT_CONFIG_FILE;
-    private static LibertyServer server = LibertyServerFactory.getLibertyServer("com.ibm.ws.security.registry.basic.fat");
+    public static LibertyServer server;
     private static final Class<?> c = FATTest.class;
     private static UserRegistryServletConnection servlet;
     private final LeakedPasswordChecker passwordChecker = new LeakedPasswordChecker(server);
 
+    @ClassRule
+    public static CheckpointRule checkpointRule = new CheckpointRule()
+                    .setConsoleLogName(FATTest.class.getSimpleName() + ".log")
+                    .setServerSetup(FATTest::serverSetUp)
+                    .setServerStart(FATTest::serverStart)
+                    .setServerTearDown(FATTest::serverTearDown);
+
     /**
      * Updates the sample, which is expected to be at the hard-coded path.
      * If this test is failing, check this path is correct.
      */
-    @BeforeClass
-    public static void setUp() throws Exception {
+    public static LibertyServer serverSetUp(ServerMode mode) throws Exception {
+        server = LibertyServerFactory.getLibertyServer("com.ibm.ws.security.registry.basic.fat");
         server.copyFileToLibertyInstallRoot("lib/features", "basicRegistryInternals-1.0.mf");
-
-        Log.info(c, "setUp", "Starting the server... (will wait for userRegistry servlet to start)");
+        Log.info(c, "serverSetUp", "Starting the server... (will wait for userRegistry servlet to start)");
         server.addInstalledAppForValidation("userRegistry");
-        startServer();
-        Log.info(c, "setUp", "Creating servlet connection the server");
+        Log.info(c, "serverSetUp", "Creating servlet connection the server");
         servlet = new UserRegistryServletConnection(server.getHostname(), server.getHttpDefaultPort());
```

Comment:
Should these two lines go after calling `serverStart()` in the `public static void serverStart(ServerMode, LibertyServer)` method?

Reply:
Right, I will update it.